services/brig/src/Brig/API/User.hs

```diff
@@ -885,48 +885,58 @@ lookupProfiles ::
   -- | The users ('others') for which to obtain the profiles.
   [UserId] ->
   AppIO [UserProfile]
-lookupProfiles self others = getSelfInfo >>= \case
-  Nothing -> pure []
-  Just selfInfo -> do
-    users <- Data.lookupUsers others >>= mapM userGC
-    css <- toMap <$> Data.lookupConnectionStatus (map userId users) [self]
-    emailVisibility' <- view (settings . emailVisibility)
-    return $ map (toProfile emailVisibility' css selfInfo) users
+lookupProfiles self others = do
+  users <- Data.lookupUsers others >>= mapM userGC
+  css <- toMap <$> Data.lookupConnectionStatus (map userId users) [self]
+  emailVisibility' <- view (settings . emailVisibility)
+  mapM (toProfile emailVisibility' css) users
```

Comment:
Now we do the same call to Galley multiple times (for every single user we want to view). :(

To stay closer to the original structure we could have a type along the lines of

```
data EmailVisible = VisibleIfOnTeam | VisibleToSelf | VisibleIfOnSameTeam (Maybe Team.TeamMember)
```

Then we do the call right in the beginning, but only in the case where it is required and pass down the result.

```
visibility <- case emailVisibility' of
  EmailVisibleIfOnTeam -> pure VisibleIfOnTeam
  EmailVisibleToSelf -> pure VisibleToSelf
  EmailVisibleIfOnSameTeam -> VisibleIfOnSameTeam <$> getViewerTeamMember u
```

It could also work without the extra type (or even by pass down a function `TeamId -> Bool` :D ), but this seems to model the idea most closely.